src/client/scripts/esm/chess/variants/variant.ts

```diff
@@ -33,6 +44,46 @@ const coaIPPromotionsAllowed = repeatPromotionsAllowedForEachColor(coaIPPromotio
 
 const gameruleModificationsOfOmegaShowcasings = { promotionRanks: null, moveRule: null, turnOrder: ['black', 'white'] }; // No promotions, no 50-move rule, and reversed turn order.
 
+interface Moveset {
+	[key: string]: {
+		individual: number[][],
+		sliding?: {
+			[key: string]: number[]
+		}
+	}
+}
+
+interface Gamerules {
```

Comment:
Same thought as above, but gamerules.js intead of movesets.js

Reply:
JsDoc does have a way to do typedefs like this; should I replace the function with a JsDoc typedef?

Reply:
Actually gamerules.js already exports the type definition by exporting the function. Why not use that?

Turns out the import path was incorrect anyway, so variant.ts wasn't even using the GameRules import. I don't know why it wasn't throwing any errors. I corrected the path, and while it still compiles, there are lots of errors now related to the GameRules type that need to be fixed.